utils/resilience-demo/README.md

```diff
@@ -0,0 +1,148 @@
+
+### Prerequisite: set up 4 separate Linux machines/VMs
+
+1. We assume Ubuntu Xenial 64-bit machines have been set up to the
+   point that they can boot, have at least one (and perhaps two)
+   unique IP addresses, and can run `apt-get` to install software
+   packages.
+
+   I have been testing these scripts using AWS EC2 instances of type
+   t3.medium + AMI: Ubuntu Server 16.04 LTS (HVM), SSD Volume Type,
+   ami-04169656fea786776 in the us-east-1 region.
+
+2. All 4 machines/VM/instances ought to have the same capacity.  You
+   can get away with provisioning more memory for what the scripts
+   call `$SERVER1`.  That machine should have 4GB of RAM.  The other
+   machines can probably work with 1GB RAM but really ought to have
+   4GB RAM also.
+
+### Prerequisite: edit the COMMON.sh script
+
+The [`COMMON.sh`](./COMMON.sh) script in this directory needs to be
+edited to reflect the IP addresses of the 4 machines/VMs that will
+execute the demos.
+
+Please see the comments embedded in the [`COMMON.sh`](./COMMON.sh)
+script for greater detail.
+
+### Prerequisites: SSH access
+
+We assume that the SSH (remote secure shell) service is available
+to all 4 machines/VMs without password prompts.  We strongly
+recommend configuring & using SSH's `$HOME/.ssh/authorized_keys`
+mechanism.
+
+All scripts assume that the demo initiator machine is *NOT* one of the
+target machines/VMs.  Therefore, all target machines/VMs must allow
+SSH session from the demo initiator machine.
+
+In some scripts, SSH is used to run commands (such as `rsync`)
+between machines/VMs, e.g., an SSH session initiated by `$SERVER1`
+to `$SERVER2`.  All target machines must be permitted to initiate SSH
+sessions to all other target machines.
+
+### Prerequisite: network firewall stuff
+
+1. We assume that the machine running these scripts may not be one of
+   the four target machines/VMs used to execute the demos. For
+   example, let's assume that you run these demo scripts on your
+   laptop in your home office network.
+
+   If you have a network firewall anywhere between your laptop and the
+   target machines/VMs must not block any traffic to or from the
+   target machines/VMs.
+
+2. If you have a firewall that affects the target machines/VMs, then
+   all traffic between all target machines must be permitted.
+
+3. The scripts are written with the assumption that the target
+   machines/VMs have a private internet LAN network.  If this is not a
+   valid assumption, then set the following values in the `COMMON.sh`
+   script.
+
+    $SERVER1=$SERVER1_EXT
+    $SERVER2=$SERVER2_EXT
+    $SERVER3=$SERVER3_EXT
+    $SERVER4=$SERVER4_EXT
+    
+### Prerequisite: Set up the runtime environment on all target machines/VMs
+
+On your laptop, run the following two commands:
+
+    ./00-setup-dev-env.sh
+    ./10-setup-slf-repo.r=on.sh
+
+# Demo: Verify that a worker restart does not disrupt sequence window verification
```

Comment:
I wouldn't say `sequence window verification` here. Maybe `does not violate state invariants`?
The sequence window verification thing is very internally specific and even internally I think it might not mean much to some folks.